### ISSUE TITLE
Make "showing selected items first" optional

### ISSUE THREAD

**Issue by brunojcm** [bonprix/vaadin-combobox-multiselect#20]
I need this for 2 reasons:

1. I need to keep alphabetical order, regardless of the selection
2. I have a container that cannot order for "synthetic" fields like 'selected'

I'm planning to work on this and provide a PR. Would you review and accept that?

**Comment by thorbenvh8** [comment]
sure, as long as default stays it gets sorted? :)

what container are you using?

**Comment by thorbenvh8** [comment]
easiest way to prevent any sorting would be adding another parameter default true and called by isSortingAllowed();

    private boolean isSortingNeeded() {
    	return isSortingAllowed() && sortingNeeded;
    }

maybe try this? :)

**Comment by brunojcm** [comment]
> what container are you using?

I'm using a "PersistentContainer" which extends AbstractBeanContainer and is a kind of a LazyQueryContainer, keeping some index of ids in memory and querying items when needed. It sorts and filter directly on the database.

> return isSortingAllowed() && sortingNeeded;

That's not exactly what would solve my problem. Actually, I need

    sortable.sort(new Object[] { ComboBoxMultiselect.SELECTED_PROPERTY, getItemCaptionPropertyId() }, new boolean[] { false, true });

to be

    sortable.sort(new Object[] { getItemCaptionPropertyId() }, new boolean[] { true });

in my case here.

**Comment by thorbenvh8** [comment]
ok can you make your pull request using an interface which default is my version and everyone can overwrite it by setting his own interface implementation? :)

do you have a documentation to the container youre using? :)

**Comment by brunojcm** [comment]
I was thinking about just inserting a new property like `showSelectedOnTop` or `orderBySelectionProperty`, with getter and setter, and do the following:

    if (getShowSelectedOnTop()) {
        sortable.sort(new Object[] { ComboBoxMultiselect.SELECTED_PROPERTY, getItemCaptionPropertyId() }, new boolean[] { false, true });
    } else {
        sortable.sort(new Object[] { getItemCaptionPropertyId() }, new boolean[] { true });
    }

What do you think?

BTW, I'm currently blocked by Issue #21, so I'm trying to figure it out first.

**Comment by thorbenvh8** [comment]
Yes that a good approach :)
like the naming and the idea :)

**Comment by thorbenvh8** [comment]
maybe only isShowSelectedOnTop() instead of get :) but yes :)

**Comment by chkpnt** [comment]
You already can set your own `Comparator`. I'm doing so, to keep the order from `BeanItemContainer` within the selected and unselected items:

```java
	public int compare(Object o1, Object o2) {
		// ...
		order = comboBox.getSelectedValueOrder(o1, o2);
		if (order != 0) {
			return order;
		}

		BeanItemContainer containerDataSource = (BeanItemContainer) comboBox.getContainerDataSource();
		int indexOfO1 = containerDataSource.indexOfId(o1);
		int indexOfO2 = containerDataSource.indexOfId(o2);
		return Integer.compare(indexOfO1, indexOfO2);
	}
```

**Comment by thorbenvh8** [comment]
yes but he is not using a container that this comparator is working for :)

**Comment by thorbenvh8** [comment]
thanks for the pr :)

**Comment by thorbenvh8** [comment]
available in version 1.1.13
